pyk/src/pyk/cterm/cterm.py

```diff
@@ -202,6 +202,8 @@ def anti_unify(
             - ``csubst2``: Constrained substitution to apply to `cterm` to obtain `other`.
         """
         new_config, self_subst, other_subst = anti_unify(self.config, other.config, kdef=kdef)
+        # todo: It's not able to distinguish between constraints in different cterms,
+        #  because variable names may be used inconsistently in different cterms.
```

Comment:
After this PR, we can further optimize `anti_unify` and provide more tests for it.